appcelerator/titanium_mobile
# fix(android): TabbedBar crash caused by setting properties before window opened

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-26758

**Summary:**
- Crash caused by settings properties "labels" or "index" before the `TabbedBar` parent window has been opened.
- Moved all property handling to `TiUITabbedBar.java` file.
- Modified "style" JS property to return default style constant if not assigned.
- Added additional property validation with warning log messages.

**Test:**
1. Build and run the below code on Android.
2. Verify that the app does not crash on startup. _(This is the fix.)_
3. Verify that 3 tabbed bars are shown onscreen.
4. Verify top bar reads `[One | Two | Three]`, where `One` is selected.
5. Verify middle bar reads `[Red | Green | Blue]`, where `Green` is selected.
6. Verify bottom bar reads `[Alpha | Beta | Charlie]`, where `Charlie` is selected.

```javascript
var window = Ti.UI.createWindow();

var tabbedBar1 = Ti.UI.createTabbedBar({
	labels: ["One", "Two", "Three"],
	index: 0,
	top: "20%",
	width: "90%",
});
window.add(tabbedBar1);

var tabbedBar2 = Ti.UI.createTabbedBar();
tabbedBar2.labels = ["Red", "Green", "Blue"];
tabbedBar2.index = 1;
tabbedBar2.top = "40%";
tabbedBar2.width = "90%";
window.add(tabbedBar2);

var tabbedBar3 = Ti.UI.createTabbedBar({
	labels: ["Uh", "Oh"],
	index: 0,
	top: "60%",
	width: "90%",
})
tabbedBar3.labels = ["Alpha", "Beta", "Charlie"];
tabbedBar3.index = 2;
window.add(tabbedBar3);

window.open();
```
